src/transformers/models/idefics/modeling_idefics.py

```diff
@@ -1103,11 +1112,16 @@ def forward(
             position_ids = position_ids.view(-1, seq_length).long()
 
         no_images = False
-        if pixel_values is None and image_embeddings is None:
-            raise ValueError("Either pixel_values and image_embeddings have to be not-None.")
-
-        elif pixel_values is not None and image_embeddings is not None:
-            raise ValueError("You cannot specify both pixel_values and image_embeddings at the same time")
+        if (
+            sum(
+                vision_input is not None
+                for vision_input in (pixel_values, image_encoder_embeddings, perceiver_embeddings)
+            )
+            != 1
+        ):
```

Comment:
Maybe just 
```suggestion
        if None not in (pixel_values, image_encoder_embeddings, perceiver_embeddings):
```

Reply:
No because I need 2 of them to be None and one to be filled

Reply:
You can also better format the Error to say which values where not None!